src/Model/Table/UsersTable.php

```diff
@@ -192,6 +197,16 @@ public function buildRules(RulesChecker $rules)
             ]);
         }
 
+        if ($this->usernameBlackList) {
```

Comment:
Please extract it into a behavior, allow to configure a callback to check the username blacklist (lines 202-204) so we will be able to configure an external method and read the blacklist from the database.

Please change name and configuration from `usernameBlacklist` to `forbiddenUsernames`.

Thanks,